src/encoded/static/libs/jsonld.js

```diff
@@ -0,0 +1,138 @@
+        identifier: context.uuid,
+        encodingFormat: 'text',
+        measurementTechnique,
+    };
+};
+
+const _mapSourceToSourceOrganization = (source) => {
+    if (!source) {
+        return {};
+    }
+
+    return {
+        name: source.title,
+        sameAs: source.url,
+        description: source.description,
+        url: `${baseUrl}${source['@id']}`,
+        identifier: source.uuid,
+    };
+};
+
+const _mapSubmitterToCreator = (submittedBy) => {
+    if (!submittedBy) {
+        return {};
+    }
+
+    return {
+        '@type': 'Person',
+        identifier: submittedBy.uuid,
+        name: submittedBy.title,
+        worksFor: {
+            '@type': 'EducationalOrganization',
+            sameAs: `${baseUrl}${submittedBy.lab}`,
+            url: `${baseUrl}${submittedBy.lab}`,
+        },
+        memberOf: !submittedBy.submits_for ? [] : submittedBy.submits_for.map(submitsFor => ({
+            '@type': 'EducationalOrganization',
+            url: `${baseUrl}${submitsFor}`,
+        })),
+    };
+};
+
+const jsonldFormatter = (context, url) => {
+    baseUrl = url || baseUrl;
+    let measurementTechnique = '';
+
+    // set technique
+    if (context.target_label) {
+        measurementTechnique = context.target_label;
+    } else if (context.assay_title) {
+        measurementTechnique = `${context.assay_title} (${context.assay_term_id})`;
+    } else if (context.annotation_type) {
+        measurementTechnique = context.annotation_type;
+    }
+
+    const mappedData = {
+        '@context': 'http://schema.org/',
+        '@type': 'Dataset',
+        url: `${baseUrl}${context['@id']}`,
+        dateCreated: context.date_created,
+        license: `${baseUrl}/help/citing-encode/`,
+        name: context.accession, // required by Google data set
+        isAccessibleForFree: true,
+        isFamilyFriendly: true,
+        identifier: context.uuid,
+        version: context.schema_version,
+        publisher: _addPublisher(context),
+        distribution: _mapContextToDistribution(context, measurementTechnique),
+        keywords: [...context.internal_tags, ...[measurementTechnique, context.accession, context.uuid, 'ENCODE', 'ENCODED', 'DCC', 'Encyclopedia of DNA Elements']],
+        measurementTechnique,
+        includedInDataCatalog: {
+            '@type': 'DataCatalog',
+            name: 'ENCODE: Encyclopedia of DNA Elements',
+            url: `${baseUrl}`,
+        },
+    };
+
+    // set description
+    if (context['@type'].some(type => type.toLowerCase() === 'annotation')) {
+        mappedData.description = context.annotation_type;
+    } else if (context['@type'].some(type => type.toLowerCase() === 'experiment' || type.toLowerCase() === 'functionalcharacterizationexperiment')) {
+        mappedData.description = context.target_label ?
+        `[${context.target_label}] ${context.assay_title} ${context.biosample_summary}`.trim() :
+        `${context.assay_title} ${context.biosample_summary}`.trim();
```

Comment:
Is it OK that `biosample_summary` often doesn’t exist (https://test.encodedcc.org/search/?type=Experiment&biosample_summary!=*)? I see on http://localhost:6543/experiments/ENCSR000INT/ that the description property is “TF ChIP-seq undefined.”

Reply:
@forresttanaka Thanks. I added a condition to replace undefined with empty string. In line 121, I added the condition on target_label and realized it after I have pushed the commit; , but I assume that is okay, though..?